src/UI/Components/LogViewer/LogViewer.tsx

```diff
@@ -0,0 +1,191 @@
+import React, { useState, useRef } from "react";
+import {
+  Button,
+  Icon,
+  Label,
+  MenuToggle,
+  Select,
+  SelectList,
+  SelectOption,
+  Toolbar,
+  ToolbarContent,
+  ToolbarGroup,
+  ToolbarItem,
+  Tooltip,
+  Flex,
+  FlexItem,
+} from "@patternfly/react-core";
+import { DownloadIcon, ExclamationCircleIcon, PauseIcon, PlayIcon } from "@patternfly/react-icons";
+import { LogViewerSearch, LogViewer as PFLogViewer } from "@patternfly/react-log-viewer";
+import { words } from "@/UI/words";
+
+export interface LogViewerData {
+  data: string[];
+  id: string;
+  name: string;
+  duration: number;
+  failed: boolean;
+}
+
+interface LogViewerProps {
+  logs: LogViewerData[];
+}
+
+/**
+ * LogViewer component displays logs with a dropdown to select log parts and a download button.
+ * Uses PatternFly's LogViewer for rendering logs.
+ * @param logs Array of log objects, each with data, id, name, duration, and failed status.
+ */
+export const LogViewerComponent: React.FC<LogViewerProps> = ({ logs }) => {
+  const [selectedLogId, setSelectedLogId] = useState(logs[0]?.id || "");
```

Comment:
shouldn't we provide empty view if there is no logs?

Reply:
The safe operator is there because we are dealing with an array, but the check is already there at a higher level to not display the logViewer if we don't have reports.